action.yml

```diff
@@ -0,0 +1,186 @@
+                }
+              }
+            }
+          }
+
+          // Sort the vulnerability rows by severity using sortOrder mapping.
+          vulnRows.sort((a, b) => {
+            const orderA = sortOrder[a.severity] || 99;
+            const orderB = sortOrder[b.severity] || 99;
+            return orderA - orderB;
+          });
+
+          const vulnCount = vulnRows.length;
+          fs.appendFileSync(process.env.GITHUB_OUTPUT, `vuln_count=${vulnCount}\n`);
+
+          if (vulnCount === 0) {
+            console.log("No vulnerabilities found. Not posting comment.");
+            fs.appendFileSync(process.env.GITHUB_OUTPUT, `comment<<EOF\n\nEOF\n`);
+            process.exit(0);
+          }
+
+          // Calculate dynamic counts
+          const criticalHighCount = vulnRows.filter(v => v.severity === 'CRITICAL' || v.severity === 'HIGH').length;
+          const totalVulnCount = vulnCount;
+
+          // Create header text with dynamic counts.
+          let headerText = `### 🔍 End of Life and Outdated JS Scan Results\n\n`;
+          if (criticalHighCount > 0) {
+            headerText += `⛔️ **Action Required:** ${criticalHighCount} Critical/High severity vulnerabilities found\n\n`;
+          }
+          headerText += `Found ${totalVulnCount} total vulnerabilities\n`;
+
+          let footerText = '\n### Remediation\n';
+          if (criticalHighCount > 0) {
+            footerText += `⚠️ **Action Required:** Critical/High severity vulnerabilities must be fixed before merging\n\n`;
+          }
+          footerText += `- Update vulnerable packages to their fixed versions where available\n`;
+          footerText += `- Run \`dependency-check\` locally to see more details\n`;
+
+          // Convert sorted objects to Markdown table rows
+          const rows = vulnRows.map(item =>
+            `| ${item.relativePath} | ${item.depName} | ${item.cve} | ${item.severityDisplay} |`
+          );
+
+          const markdownTable = tableHeader + rows.join('\n');
+          const commentBody = `${headerText}${markdownTable}${footerText}`;
+
+          fs.appendFileSync(process.env.GITHUB_OUTPUT, `comment<<EOF\n${commentBody}\nEOF\n`);
+
+    - name: Find existing dependency check comment
+      if: ${{ steps.generate_comment.outputs.vuln_count != '0' && !cancelled() }}
+      id: find_comment
+      uses: peter-evans/find-comment@v3
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        body-includes: 'End of Life and Outdated JS Scan Results'
+        comment-author: 'github-actions[bot]'
+
+    - name: Create dependency check comment
+      if: ${{ steps.generate_comment.outputs.vuln_count != '0' && steps.find_comment.outputs.comment-id == '' && !cancelled() }}
+      uses: peter-evans/create-or-update-comment@v4
+      with:
+        token: ${{ inputs.token }}
+        issue-number: ${{ github.event.pull_request.number }}
+        body: ${{ steps.generate_comment.outputs.comment }}
```

Comment:
Why do we need these 2 ?

Reply:
This is to avoid putting comments on each commit push. A comment will be added only once, and in later commits the existing comment will be edited.